custom_components/ha_creality_ws/__init__.py

```
@@ -468,12 +488,163 @@ async def request_cfs_info(call: ServiceCall) -> None:
+            if state in BUSY_PRINT_STATES:
+                raise ServiceValidationError(
+                    f"{coord.client.host} is {state}; refusing to change CFS "
+                    "material while the printer is busy."
+                )
+
+        for coord in targets:
+            host = coord.client.host
+            try:
+                _LOGGER.info("Sending modifyMaterial to %s: %s", host, payload)
+                await coord.client.send_set_retry(modifyMaterial=payload)
+            except Exception as exc:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)